sdks/java/extensions/sbe/src/main/java/org/apache/beam/sdk/extensions/sbe/PrimitiveSbeField.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sbe;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.sdk.extensions.sbe.UnsignedOptions.Behavior;
+import org.apache.beam.sdk.schemas.Schema.Field;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import uk.co.real_logic.sbe.PrimitiveType;
+
+/** Represents a primitive SBE field. */
+@Experimental(Kind.SCHEMAS)
+@AutoValue
+public abstract class PrimitiveSbeField implements SbeField {
+  private static final long serialVersionUID = 1L;
+
+  @Override
+  public abstract String name();
+
+  @Override
+  public abstract Boolean isRequired();
+
+  public abstract PrimitiveType type();
+
+  @Override
+  public Field asBeamField(SbeFieldOptions options) {
+    FieldType type = beamType(options);
+    return isRequired() ? Field.of(name(), type) : Field.nullable(name(), type);
+  }
+
+  private FieldType beamType(SbeFieldOptions options) {
+    switch (type()) {
+      case CHAR:
+        // TODO(BEAM-12697): Support char types
+        throw new UnsupportedOperationException("char type not supported yet");
```

Comment:
nit: could be good to reference the jira in the error message so users have somewhere to report they've run into this.
```suggestion
        // TODO(BEAM-12697): Support char types
        throw new UnsupportedOperationException("char type not supported yet (BEAM-12697)");
```

Reply:
Done.